From 3ca6ccd4970704de1ab1286528b035e170aa7931 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:09 -0400
Subject: [ethernet] qlcnic: add missing destroy_workqueue() on error path in qlcnic_probe()

Message-id: <1408626422-3941-26-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88553
O-Subject: [PATCH RHEL 7.1 025/138] qlcnic: add missing destroy_workqueue() on error path in qlcnic_probe()
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 2b1f18a4d6ae8057a93e736a34cdcca925279403

Add the missing destroy_workqueue() before return from
qlcnic_probe() in the error handling case.

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index adcb915..63b5b03 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2258,7 +2258,7 @@ qlcnic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 
  err = qlcnic_alloc_adapter_resources(adapter);
  if (err)
-  goto err_out_free_netdev;
+  goto err_out_free_wq;
 
  adapter->dev_rst_time = jiffies;
  adapter->ahw->revision_id = pdev->revision;
@@ -2397,6 +2397,9 @@ err_out_disable_msi:
 err_out_free_hw:
  qlcnic_free_adapter_resources(adapter);
 
+err_out_free_wq:
+ destroy_workqueue(adapter->qlcnic_wq);
+
 err_out_free_netdev:
  free_netdev(netdev);
 
-- 
1.7.1